components/annot-tool-group.vue

```diff
@@ -0,0 +1,323 @@
+                }
+
+                // 3. Trigger a save of this transformation to the annotated table in the store
+                // NOTE: 'transformHeuristics' are currently not being saved to the store
+                this.$emit("update:dataTable", {
+
+                    // transformHeuristics: this.subjectAvailability,
+                    transformedTable: transformedTable
+                });
+            },
+
+            declareMissing(p_row) {
+
+                // 1. Indicate all subjects with this value are unavailable
+                this.setSubjectAvailability(p_row.column_name, p_row.raw_value, false);
+
+                // 2. Send missing value to store
+                this.$emit("update:missingValue", {
+
+                    column: p_row.column_name,
+                    value: p_row.raw_value
+                });
+            },
+
+            initalizeAvailabilityMap() {
+
+                // 0. Wipe the availability map
+                this.subjectAvailability = {};
+
+                // 1. All subjects are initially available for this tool group for annotation
+                for ( const row of this.dataTable.original ) {
+
+                    this.subjectAvailability[row[this.idField]] = {};
+
+                    for ( const toolName of this.relevantColumns ) {
+
+                        this.subjectAvailability[row[this.idField]][toolName] = true;
+                    }
+                }
+            },
+
+            refreshTableItems() {
+
+                // 0. Wipe table contents
+                this.tableItems = {};
+                for ( const itemName of this.relevantColumns ) {
+                    this.tableItems[itemName] = [];
+                }
+
+                // 1. Make a row for each column value
+                for ( const columnName in this.uniqueValues ) {
+
+                    for ( const rawValue of this.uniqueValues[columnName] ) {
+
+                        // Only display values that are not declared as missing
+                        if ( !this.isMissingValue(columnName, rawValue) ) {
+
+                            // A. Get the value description from the data dictionary, if available
+                            const valueDescription = this.valueDescription(columnName, rawValue);
+
+                            // B. Save the new row entry
+                            this.tableItems[columnName].push({
+
+                                column_name: columnName,
+                                description: valueDescription,
+                                raw_value: rawValue
+                            });
+
+                            // C. Update the subject availability map as needed for this value
+                            this.setSubjectAvailability(columnName, rawValue, true);
+                        }
+                    }
+                }
+            },
+
+            setSubjectAvailability(p_columnName, p_rawValue, p_availabilityStatus) {
+
```

Comment:
Also commenting for the future here. Currently, a subject is not available for the current tool if ANY of its values in the relevant columns are "missing". The current implementation works, but probably needs to be rewritten once we allow for other availability "rules" that have to take into consideration of the other columns (e.g. if ALL have to be missing for a subject to be not available).

Reply:
Created an issue for this here: https://github.com/neurobagel/annotation_tool/issues/161